### PR TITLE
Changes to operator hub tooling for 2.9 release

### PR DESCRIPTION
**Pull request by pebrc**
I realised while looking at https://github.com/k8s-operatorhub/community-operators/tree/main/operators/elastic-cloud-eck/2.8.0/metadata that the format for the community operator submissions has changed to bundle format. 

I am not sure if a submission is still possible in legacy package format once the previously published versions have been (auto-) converted to bundles.

This change 
* moves the manifest generation to bundle format 
* removes all the "extra" functions in the bundle PR submissions code 
  * renaming of the CSV is not necessary if we just don't generate the version into the name to begin with. The [registry seems to look at `Kind`](https://github.com/operator-framework/operator-registry/blob/82a91fa2bf806d06c11454e1123cc97e60aed22a/pkg/registry/csv.go#L113) anyway to identify CSVs
  * the image SHA replacement code was dead and duplicating the working code in the manifest generation
  * the `*package.yaml` thingy is no longer needed in bundle format
* removes the `bundle generate` command which is not necessary if we template in bundle format to start with. 

Tested locally.

### CODE REVIEW

**Comment by naemono** [comment]
@pebrc I'm doing some local testing/verification, but the the initial thing I noticed is that the readme in `hack/operatorhub` needs updated to remove references to `bundle generate`.

**Comment by pebrc** [comment]
@naemono good point forgot about that one!

**Comment by naemono** [comment]
@pebrc so testing locally and for the `community-operators` I see the `manifests` directory missing, which seems to contain all of the yaml manifests for the previous release, and note the missing `metadata/annotations.yaml` file.

```
❯ pwd
/var/folders/6j/5760jrwd7d9_3mz7rxxxzz0r0000gn/T/eck-redhat-operations387263295/community-operators/operators/elastic-cloud-eck

❯ find 2.9.0-bc8
2.9.0-bc8
2.9.0-bc8/elastic-cloud-eck.v2.9.0-bc8.clusterserviceversion.yaml
2.9.0-bc8/agents.agent.k8s.elastic.co.crd.yaml
2.9.0-bc8/kibanas.kibana.k8s.elastic.co.crd.yaml
2.9.0-bc8/apmservers.apm.k8s.elastic.co.crd.yaml
2.9.0-bc8/elasticmapsservers.maps.k8s.elastic.co.crd.yaml
2.9.0-bc8/logstashes.logstash.k8s.elastic.co.crd.yaml
2.9.0-bc8/beats.beat.k8s.elastic.co.crd.yaml
2.9.0-bc8/elasticsearchautoscalers.autoscaling.k8s.elastic.co.crd.yaml
2.9.0-bc8/elasticsearches.elasticsearch.k8s.elastic.co.crd.yaml
2.9.0-bc8/enterprisesearches.enterprisesearch.k8s.elastic.co.crd.yaml
2.9.0-bc8/stackconfigpolicies.stackconfigpolicy.k8s.elastic.co.crd.yaml

❯ find 2.8.0
2.8.0
2.8.0/manifests
2.8.0/manifests/elastic-cloud-eck.v2.8.0.clusterserviceversion.yaml
2.8.0/manifests/agents.agent.k8s.elastic.co.crd.yaml
2.8.0/manifests/kibanas.kibana.k8s.elastic.co.crd.yaml
2.8.0/manifests/apmservers.apm.k8s.elastic.co.crd.yaml
2.8.0/manifests/elasticmapsservers.maps.k8s.elastic.co.crd.yaml
2.8.0/manifests/logstashes.logstash.k8s.elastic.co.crd.yaml
2.8.0/manifests/beats.beat.k8s.elastic.co.crd.yaml
2.8.0/manifests/elasticsearchautoscalers.autoscaling.k8s.elastic.co.crd.yaml
2.8.0/manifests/elasticsearches.elasticsearch.k8s.elastic.co.crd.yaml
2.8.0/manifests/enterprisesearches.enterprisesearch.k8s.elastic.co.crd.yaml
2.8.0/manifests/stackconfigpolicies.stackconfigpolicy.k8s.elastic.co.crd.yaml
2.8.0/metadata
2.8.0/metadata/annotations.yaml
```

I am certainly no expert in this bundle format, so I wonder if both the missing annotations file, and the missing directory where the yaml manifests live will cause any issues? Do you know?

**Comment by naemono** [comment]
Also it fails for the certified-operators portion when preparing to create the pull requests:

```
2023/07/25 09:26:40 copying (/var/folders/6j/5760jrwd7d9_3mz7rxxxzz0r0000gn/T/eck-redhat-operations387263295/certified-operators/operators/elasticsearch-eck-operator-certified/2.9.0-bc8/manifests/elasticsearch-eck-operator-certified.v2.9.0-bc8.clusterserviceversion.yaml) to (/var/folders/6j/5760jrwd7d9_3mz7rxxxzz0r0000gn/T/eck-redhat-operations387263295/certified-operators/operators/elasticsearch-eck-operator-certified/2.9.0-bc8/manifests/elasticsearch-eck-operator-certified.clusterserviceversion.yaml)

Error: while copying (/var/folders/6j/5760jrwd7d9_3mz7rxxxzz0r0000gn/T/eck-redhat-operations387263295/certified-operators/operators/elasticsearch-eck-operator-certified/2.9.0-bc8/manifests/elasticsearch-eck-operator-certified.v2.9.0-bc8.clusterserviceversion.yaml) to (/var/folders/6j/5760jrwd7d9_3mz7rxxxzz0r0000gn/T/eck-redhat-operations387263295/certified-operators/operators/elasticsearch-eck-operator-certified/2.9.0-bc8/manifests/elasticsearch-eck-operator-certified.clusterserviceversion.yaml)
```

This is because all of the yaml manifests are not in the `manifests` subdirectory initially:

```
❯ pwd
/var/folders/6j/5760jrwd7d9_3mz7rxxxzz0r0000gn/T/eck-redhat-operations387263295/certified-operators/operators/elasticsearch-eck-operator-certified/2.9.0-bc8

❯ ls
agents.agent.k8s.elastic.co.crd.yaml                                       beats.beat.k8s.elastic.co.crd.yaml                                         elasticsearch-eck-operator-certified.v2.9.0-bc8.clusterserviceversion.yaml elasticsearches.elasticsearch.k8s.elastic.co.crd.yaml                      kibanas.kibana.k8s.elastic.co.crd.yaml                                     stackconfigpolicies.stackconfigpolicy.k8s.elastic.co.crd.yaml
apmservers.apm.k8s.elastic.co.crd.yaml                                     elasticmapsservers.maps.k8s.elastic.co.crd.yaml                            elasticsearchautoscalers.autoscaling.k8s.elastic.co.crd.yaml               enterprisesearches.enterprisesearch.k8s.elastic.co.crd.yaml                logstashes.logstash.k8s.elastic.co.crd.yaml
```

**Comment by pebrc** [comment]
Interesting @naemono let me test that again. I thought I had done a couple of runs last night including the PR creation in `--dry-run` mode

**Comment by pebrc** [comment]
I cannot reproduce your issue I have: 

```
certified-operators/
└── 2.8.0
    ├── manifests
    │   ├── agents.agent.k8s.elastic.co.crd.yaml
    │   ├── apmservers.apm.k8s.elastic.co.crd.yaml
    │   ├── beats.beat.k8s.elastic.co.crd.yaml
    │   ├── elasticmapsservers.maps.k8s.elastic.co.crd.yaml
    │   ├── elasticsearch-eck-operator-certified.clusterserviceversion.yaml
    │   ├── elasticsearchautoscalers.autoscaling.k8s.elastic.co.crd.yaml
    │   ├── elasticsearches.elasticsearch.k8s.elastic.co.crd.yaml
    │   ├── enterprisesearches.enterprisesearch.k8s.elastic.co.crd.yaml
    │   ├── kibanas.kibana.k8s.elastic.co.crd.yaml
    │   ├── logstashes.logstash.k8s.elastic.co.crd.yaml
    │   └── stackconfigpolicies.stackconfigpolicy.k8s.elastic.co.crd.yaml
    └── metadata
        └── annotations.yaml

4 directories, 12 files
17:08~/s/c/h/operatorhub (operatorhub-changes-2.9)> tree community-operators/
community-operators/
└── 2.8.0
    ├── manifests
    │   ├── agents.agent.k8s.elastic.co.crd.yaml
    │   ├── apmservers.apm.k8s.elastic.co.crd.yaml
    │   ├── beats.beat.k8s.elastic.co.crd.yaml
    │   ├── elastic-cloud-eck.clusterserviceversion.yaml
    │   ├── elasticmapsservers.maps.k8s.elastic.co.crd.yaml
    │   ├── elasticsearchautoscalers.autoscaling.k8s.elastic.co.crd.yaml
    │   ├── elasticsearches.elasticsearch.k8s.elastic.co.crd.yaml
    │   ├── enterprisesearches.enterprisesearch.k8s.elastic.co.crd.yaml
    │   ├── kibanas.kibana.k8s.elastic.co.crd.yaml
    │   ├── logstashes.logstash.k8s.elastic.co.crd.yaml
    │   └── stackconfigpolicies.stackconfigpolicy.k8s.elastic.co.crd.yaml
    └── metadata
        └── annotations.yaml

```

generated by `go run main.go generate-manifests --conf=config.yaml --enable-vault=false -y ../../config/crds.yaml -y ../../config/operator.yaml -a <API_KEY> -p <PROJECT_ID>`

with the following config: 

```
newVersion: 2.8.0
prevVersion: 2.7.0
stackVersion: 8.8.0
crds:
  - name: elasticsearches.elasticsearch.k8s.elastic.co
    displayName: Elasticsearch Cluster
    description: Instance of an Elasticsearch cluster
  - name: elasticsearchautoscalers.autoscaling.k8s.elastic.co
    displayName: Elasticsearch Autoscaler
    description: Instance of an Elasticsearch autoscaler
  - name: kibanas.kibana.k8s.elastic.co
    displayName: Kibana
    description: Kibana instance
  - name: apmservers.apm.k8s.elastic.co
    displayName: APM Server
    description: APM Server instance
  - name: enterprisesearches.enterprisesearch.k8s.elastic.co
    displayName: Enterprise Search
    description: Enterprise Search instance
  - name: beats.beat.k8s.elastic.co
    displayName: Beats
    description: Beats instance
  - name: agents.agent.k8s.elastic.co
    displayName: Elastic Agent
    description: Elastic Agent instance
  - name: elasticmapsservers.maps.k8s.elastic.co
    displayName: Elastic Maps Server
    description: Elastic Maps Server instance
  - name: stackconfigpolicies.stackconfigpolicy.k8s.elastic.co
    displayName: Elastic Stack Config Policy
    description: Elastic Stack Config Policy
  - name: logstashes.logstash.k8s.elastic.co
    displayName: Logstash
    description: Logstash instance
packages:
  - outputPath: community-operators
    packageName: elastic-cloud-eck
    distributionChannel: community-operators
    operatorRepo: docker.elastic.co/eck/eck-operator
    ubiOnly: false
  - outputPath: certified-operators
    packageName: elasticsearch-eck-operator-certified
    distributionChannel: certified-operators
    minSupportedOpenshiftVersion: v4.9
    operatorRepo: registry.connect.redhat.com/elastic/eck-operator
    ubiOnly: true
    digestPinning: true
```

**Comment by naemono** [comment]
@pebrc somehow my local branch changed from `operatorhub-changes-2.9` => `main` on me. I'm verifying one more time, my apologies.

**Comment by naemono** [comment]
Ok @pebrc I see what I was expecting in the temporary directory now:
```
❯ pwd
/var/folders/6j/5760jrwd7d9_3mz7rxxxzz0r0000gn/T/eck-redhat-operations2676362260

❯ tree certified-operators/operators/elasticsearch-eck-operator-certified/2.9.0-bc8
certified-operators/operators/elasticsearch-eck-operator-certified/2.9.0-bc8
├── manifests
│   ├── agents.agent.k8s.elastic.co.crd.yaml
│   ├── apmservers.apm.k8s.elastic.co.crd.yaml
│   ├── beats.beat.k8s.elastic.co.crd.yaml
│   ├── elasticmapsservers.maps.k8s.elastic.co.crd.yaml
│   ├── elasticsearch-eck-operator-certified.clusterserviceversion.yaml
│   ├── elasticsearchautoscalers.autoscaling.k8s.elastic.co.crd.yaml
│   ├── elasticsearches.elasticsearch.k8s.elastic.co.crd.yaml
│   ├── enterprisesearches.enterprisesearch.k8s.elastic.co.crd.yaml
│   ├── kibanas.kibana.k8s.elastic.co.crd.yaml
│   ├── logstashes.logstash.k8s.elastic.co.crd.yaml
│   └── stackconfigpolicies.stackconfigpolicy.k8s.elastic.co.crd.yaml
└── metadata
    └── annotations.yaml

3 directories, 12 files

❯ tree community-operators/operators/elastic-cloud-eck/2.9.0-bc8
community-operators/operators/elastic-cloud-eck/2.9.0-bc8
├── manifests
│   ├── agents.agent.k8s.elastic.co.crd.yaml
│   ├── apmservers.apm.k8s.elastic.co.crd.yaml
│   ├── beats.beat.k8s.elastic.co.crd.yaml
│   ├── elastic-cloud-eck.clusterserviceversion.yaml
│   ├── elasticmapsservers.maps.k8s.elastic.co.crd.yaml
│   ├── elasticsearchautoscalers.autoscaling.k8s.elastic.co.crd.yaml
│   ├── elasticsearches.elasticsearch.k8s.elastic.co.crd.yaml
│   ├── enterprisesearches.enterprisesearch.k8s.elastic.co.crd.yaml
│   ├── kibanas.kibana.k8s.elastic.co.crd.yaml
│   ├── logstashes.logstash.k8s.elastic.co.crd.yaml
│   └── stackconfigpolicies.stackconfigpolicy.k8s.elastic.co.crd.yaml
└── metadata
    └── annotations.yaml

3 directories, 12 files
```

Finalizing verification of code changes now...

File: hack/operatorhub/internal/operatorhub/operatorhub.go [hunk 1]
```diff
@@ -502,35 +503,38 @@ func render(params *RenderParams, templatesDir, outDir string) error {
...
 	}
 
-	if err := renderCRDs(params, versionDir); err != nil {
+	if err := renderCRDs(params, manifestDir); err != nil {
 		return err
 	}
 
-	// package file is written outside the version directory
-	return renderPackageFile(params, templatesDir, outDir)
+	// annotations file is written to a separate directory called metadta
```
Comment (naemono) [inline hack/operatorhub/internal/operatorhub/operatorhub.go:525]:
```suggestion
	// annotations file is written to a separate directory called metadata
```

**Review by naemono — APPROVED** [review]
Looks great, with one tiny typo. 👍

**Comment by pebrc** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|2.9|[](https://github.com/elastic/cloud-on-k8s/pull/7059)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**